egress_proxy/prepare-proxy.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+# Exit if any step fails
+set -e
+
+eval "$(jq -r '@sh "GITREF=\(.gitref)"')"
```

Comment:
Oooo this is nifty, the variable declaration.

Is `jq` picking up stdin? how does that work?

Reply:
That's my understanding, but I'll be honest I just copied that from the FAC implementation and didn't bother looking too deeply into it because it's working.

Reply:
Yes, the JSON comes in via stdin... See the [documentation on the `data.external` data source's "protocol"](https://registry.terraform.io/providers/hashicorp/external/latest/docs/data-sources/external#external-program-protocol)!

Reply:
The bit I was more unsure about—in case anyone knows more about this—is the pattern of "some command in a bash script picks up stdin without it being specified".